.github/workflows/release.yml

```diff
@@ -0,0 +1,32 @@
+name: pre-release-binaries
+on:
+    push:
+      tags:
+        - "v*.*.*"
+
+jobs:
+  build:
+    name: Build and Publish
+    runs-on: ubuntu-latest
+    if: github.event_name != 'pull_request'
+    strategy:
+      matrix:
+        goos: [linux, darwin]
+        goarch: [amd64, arm64]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+      - name: Build
+        run: |
+          go run gotest.tools/gotestsum@latest --format github-actions ./...
+          mkdir -p dist
+          GOOS=${{ matrix.goos }} GOARCH=${{ matrix.goarch }} go build -o dist/cofidectl-${{ matrix.goos }}-${{ matrix.goarch }} ./cmd/cofidectl/main.go
+      - name: Create Release
+        uses: softprops/action-gh-release@v2
+        with:
+            prerelease: true
```

Comment:
What would cause us to not want to release a prerelease, and how often would it happen? Shall we just release? Not the end of the world if we yank it at this early stage.

Reply:
I would have thought most of the time we want "release" to be a more deliberate thing than a tag that we need just for uptake/ci for other consuming projects...

Reply:
Sounds fun!